libs/metamask/src/lib/+state/metamask.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@angular/core';
+import { providers, utils } from 'ethers'
+
+@Injectable({ providedIn: 'root' })
+export class MetamaskService {
+
+  private provider: providers.Web3Provider;
+  private account: string = '';
+  private signer?: providers.JsonRpcSigner;
+
+  private get ethereum() {
+    if (typeof window !== 'undefined' && 'ethereum' in window) {
+      const ethereum = (window as any).ethereum;
+      if (!ethereum) throw new Error('no-web3-provider');
+      if (!ethereum.isMetaMask) throw new Error('not-metamask');
+      return ethereum;
+    }
+  }
+
+  constructor() {
+    this.provider = new providers.Web3Provider(this.ethereum);
+  }
+
+  async getAccount() {
+    if (this.account) return this.account;
+    const accounts = await this.ethereum.request({ method: 'eth_accounts' });
+
+    if (!!accounts && accounts.length) {
+      this.account = accounts[0];
+      this.signer = this.provider.getSigner();
+    }
+
+    return this.account;
```

Comment:
This will return undefined if there is no account or if user don't allow, instead of throw an error. Is it the expected behavior ?